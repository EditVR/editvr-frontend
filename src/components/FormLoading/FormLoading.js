/**
 * @file FormLoading.js
 * Exports a component that renders an indeterminate loading spinner for forms.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress, withStyles } from '@material-ui/core';

import FormLoadingStyles from './FormLoading.style';

const FormLoading = ({ classes: { spinner } }) => (
  <CircularProgress className={spinner} size={32} />
);

FormLoading.propTypes = {
  classes: PropTypes.shape({
    spinner: PropTypes.string.isRequired
  }).isRequired
};

export default withStyles(FormLoadingStyles)(FormLoading);
