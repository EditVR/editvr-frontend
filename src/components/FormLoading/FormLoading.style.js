/**
 * @file FormLoading.style.js
 * Exports FormLoading component styles.
 */

export default theme => ({
  spinner: {
    float: 'right',
    marginTop: theme.spacing.unit * 3
  }
});
