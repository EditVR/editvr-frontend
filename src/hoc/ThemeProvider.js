/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
