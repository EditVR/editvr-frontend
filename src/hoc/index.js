/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider';

export { ThemeProvider };
