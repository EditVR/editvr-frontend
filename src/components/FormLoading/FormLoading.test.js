/**
 * @file FormLoading.test.js
 * Contains tests for FormLoading.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '../../hoc';
import FormLoading from './FormLoading';

const ThemedFormLoading = ThemeProvider(FormLoading);

describe('<FormLoading />', () => {
  it('Matches its snapshot', () => {
    expect(renderer.create(<ThemedFormLoading />).toJSON()).toMatchSnapshot();
  });
});
