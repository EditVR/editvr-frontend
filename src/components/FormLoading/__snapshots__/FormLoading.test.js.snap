// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLoading /> Matches its snapshot 1`] = `
<div
  className="MuiCircularProgress-root-2 MuiCircularProgress-colorPrimary-5 MuiCircularProgress-indeterminate-4 FormLoading-spinner-1"
  role="progressbar"
  style={
    Object {
      "height": 32,
      "width": 32,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg-7"
    viewBox="0 0 50 50"
  >
    <circle
      className="MuiCircularProgress-circle-8 MuiCircularProgress-circleIndeterminate-10"
      cx={25}
      cy={25}
      fill="none"
      r={20}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;
